### PR TITLE
LUPA example

### PR DESCRIPTION
**Pull request by cmichelenstrofer**
## Description
Add a third tutorial based on the LUPA device (https://pmec-osu.github.io/LUPA/). 

## Type of PR
- [ ] Bug fix
- [ ] New feature
- [x] Documentation
- [ ] Other: (specify)

## Checklist for PR
- [x] Authors read the [contribution guidelines](https://github.com/SNL-WaterPower/WecOptTool/blob/main/.github/CONTRIBUTING.md)
- [x] The pull request is from an issue branch (not main) on *your* fork, to the [main branch in WecOptTool](https://github.com/SNL-WaterPower/WecOptTool).
- [x] The authors have given the admins edit access
- [x] All changes adhere to the style guide including PEP8, Docstrings, and Type Hints.
- [x] Modified the documentation if applicable
- [x] Modified or added a new test
- [x] All tests pass
- [x] [Reference or close any relevant issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)

## Additional details
Compared to the first two tutorials this tutorial adds:
* multiple DOFs
* multi-body WEC
* non-identity PTO kinematics matrix, that depends on multiple WEC DOFs
* irregular waves
* mooring forces
* more complex and realistic constraints (e.g. generator continuous torque)

### CODE REVIEW

File: examples/tutorial_3_LUPA.ipynb
Comment (ryancoe) [inline examples/tutorial_3_LUPA.ipynb:1]: It seems like getting equal aspect ratios in a 3D plot in `matplotlib` is not so trivial, but we need to do one of the following:

1. 3D equal aspect ratios
2. Plot just a 2D cross section (wouldn't be able to see mesh)
3. Explain in markdown that the aspect ratio is wrong

image

[the remark this answers is not in the document]
Comment (michaelcdevin) [reply examples/tutorial_3_LUPA.ipynb:1]: Thanks, the heading levels are all fixed now.

[the remark this answers is not in the document]
Comment (michaelcdevin) [reply examples/tutorial_3_LUPA.ipynb:1]: Negative is now removed to match the convention used elsewhere in the tutorial

[the remark this answers is not in the document]
Comment (michaelcdevin) [reply examples/tutorial_3_LUPA.ipynb:1]: Good call, I combined these into one plot with twin y axes. With more/wider frequencies, both the North and South sites show the full spectrum, so I think the markdown is okay (I did modify the markdown a bit to discuss the y axis scaling though)

Comment (michaelcdevin) [reply examples/tutorial_3_LUPA.ipynb:1]: The `ax.set_box_aspect(aspect = (xscale, yscale, zscale))` command for matplotlib.pyplot.Axes() is really useful here. I added that to the spar and combined meshes so they aren't "flattened" like this :)

[the remark this answers is not in the document]
Comment (michaelcdevin) [reply examples/tutorial_3_LUPA.ipynb:1]: Thanks, the Part 1 results are plotted similarly to Tutorial 2, with modifications made with respect to elements highlighted by Tutorial 3 (different constraints, multiple DOFs, etc.)

**Review by michaelcdevin — APPROVED** [review]

**Pull request merged**